failed to list alerting records

Might be caused by #804 

```java
org.springframework.jdbc.BadSqlGrammarException: jOOQ; bad SQL grammar [select count(*) "c" from (select "bithon_alert_record"."record_id", "bithon_alert_record"."alert_id", "bithon_alert_record"."alert_name", "bithon_alert_record"."app_name", "bithon_alert_record"."created_at" from "bithon_alert_record" where "bithon_alert_record"."alert_id" = ?) "t" ("v0", "v1", "v2", "v3", "v4")]
	at org.jooq_3.17.26.CLICKHOUSE.debug(Unknown Source)
	at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:112)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:107)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:116)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:116)
	at org.springframework.boot.autoconfigure.jooq.DefaultExceptionTranslatorExecuteListener.handle(DefaultExceptionTranslatorExecuteListener.java:93)
	at org.springframework.boot.autoconfigure.jooq.DefaultExceptionTranslatorExecuteListener.exception(DefaultExceptionTranslatorExecuteListener.java:78)
	at org.jooq.impl.ExecuteListeners.exception(ExecuteListeners.java:276)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:345)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:301)
	at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:322)
	at org.jooq.impl.ResultQueryTrait.fetchOne(ResultQueryTrait.java:509)
	at org.jooq.impl.DefaultDSLContext.fetchCount(DefaultDSLContext.java:4753)
	at org.bithon.server.storage.jdbc.alerting.AlertRecordJdbcStorage.getAlertRecords(AlertRecordJdbcStorage.java:129)
	at org.bithon.server.alerting.manager.api.AlertQueryApi.getRecordList(AlertQueryApi.java:200)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:255)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:188)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:926)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:831)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:914)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:590)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885)
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.bithon.agent.sentinel.servlet.filter.jakarta.JakartaServletFilter.doFilter(JakartaServletFilter.java:61)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.bithon.server.starter.config.UrlHandlerFilter.doFilterInternal(UrlHandlerFilter.java:120)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:108)
	at org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:231)
	at org.springframework.security.web.ObservationFilterChainDecorator$FilterObservation$SimpleFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:479)
	at org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$1(ObservationFilterChainDecorator.java:340)
	at org.springframework.security.web.ObservationFilterChainDecorator.lambda$wrapSecured$0(ObservationFilterChainDecorator.java:82)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:128)
	at org.springframework.security.web.access.intercept.AuthorizationFilter.doFilter(AuthorizationFilter.java:100)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:131)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:85)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:100)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:179)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:227)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:221)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter.doFilterInternal(OAuth2AuthorizationRequestRedirectFilter.java:198)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.bithon.server.webapp.security.JwtAuthenticationFilter.doFilterInternal(JwtAuthenticationFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:91)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:82)
	at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:69)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240)
	at org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$0(ObservationFilterChainDecorator.java:323)
	at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:224)
	at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:233)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:191)
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)
	at org.springframework.web.servlet.handler.HandlerMappingIntrospector.lambda$createCacheFilter$3(HandlerMappingIntrospector.java:195)
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113)
	at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:74)
	at org.springframework.security.config.annotation.web.configuration.WebMvcSecurityConfiguration$CompositeFilterChainProxy.doFilter(WebMvcSecurityConfiguration.java:230)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:352)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:268)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:107)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:731)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:389)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:904)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.sql.BatchUpdateException: Code: 62, e.displayText() = DB::Exception: Syntax error: failed at position 316 ('('): ("v0", "v1", "v2", "v3", "v4"). Expected one of: JOIN, SETTINGS, WITH, GROUP BY, WINDOW, ORDER BY, UNION, FINAL, ARRAY JOIN, WHERE, FORMAT, HAVING, INNER, LIMIT, OFFSET, PREWHERE, SAMPLE, INTO OUTFILE, array join, LEFT ARRAY JOIN, end of query, table, table function, subquery or list of joined tables (version vSClickhouse-21.8.4.1-009)

	at com.clickhouse.jdbc.SqlExceptionUtils.batchUpdateError(SqlExceptionUtils.java:107)
	at com.clickhouse.jdbc.internal.SqlBasedPreparedStatement.executeAny(SqlBasedPreparedStatement.java:219)
	at com.clickhouse.jdbc.internal.SqlBasedPreparedStatement.execute(SqlBasedPreparedStatement.java:430)
	at com.alibaba.druid.pool.DruidPooledPreparedStatement.execute(DruidPooledPreparedStatement.java:497)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
	at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4692)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:236)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:330)
	... 156 more
```
```sql
select
  count(*) "c"
from
  (
    select
      "bithon_alert_record"."record_id",
      "bithon_alert_record"."alert_id",
      "bithon_alert_record"."alert_name",
      "bithon_alert_record"."app_name",
      "bithon_alert_record"."created_at"
    from
      "bithon_alert_record"
    where
      "bithon_alert_record"."alert_id" = ?
  ) "t" ("v0", "v1", "v2", "v3", "v4")
```

--- a/dev/forbidden-apis-bithon.txt
+++ b/dev/forbidden-apis-bithon.txt
@@ -16,7 +16,10 @@ java.util.concurrent.Executors#newSingleThreadScheduledExecutor(java.util.concur
 java.util.concurrent.Executors#newScheduledThreadPool(int)
 java.util.concurrent.Executors#newScheduledThreadPool(int, java.util.concurrent.ThreadFactory)
 
- @defaultMessage The fetchExists generates 'exists' sub-queries which is not supported by ClickHouse. Use fetchCount instead
+@defaultMessage The fetchExists generates 'exists' sub-queries which is not supported by ClickHouse. Use fetchCount instead
 org.jooq.DSLContext#fetchExists(org.jooq.Select)
 org.jooq.DSLContext#fetchExists(org.jooq.Table)
 org.jooq.DSLContext#fetchExists(org.jooq.Table, org.jooq.Condition)
+
+@defaultMessage The fetchCount generates wrong SQL, use other fetchCount overrides instead
+org.jooq.DSLContext#fetchCount(org.jooq.Select)

--- a/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/AlertObjectJdbcStorage.java
+++ b/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/AlertObjectJdbcStorage.java
@@ -39,6 +39,7 @@ import org.bithon.server.storage.jdbc.JdbcStorageProviderConfiguration;
 import org.bithon.server.storage.jdbc.common.dialect.ISqlDialect;
 import org.bithon.server.storage.jdbc.common.dialect.SqlDialectManager;
 import org.bithon.server.storage.jdbc.common.jooq.Tables;
+import org.jooq.Condition;
 import org.jooq.DSLContext;
 import org.jooq.Field;
 import org.jooq.Record;
@@ -77,7 +78,7 @@ public class AlertObjectJdbcStorage implements IAlertObjectStorage {
              sqlDialectManager.getSqlDialect(storageProvider.getDslContext()).quoteIdentifier(Tables.BITHON_ALERT_STATE.getName()),
              objectMapper,
              storageConfig
-            );
+        );
     }
 
     public AlertObjectJdbcStorage(DSLContext dslContext,
@@ -126,18 +127,16 @@ public class AlertObjectJdbcStorage implements IAlertObjectStorage {
         // use fetchCount instead of fetchExists
         // because the former one uses 'exists' subclause in the generated SQL,
         // and for ClickHouse, this clause is not recognizable
-        return dslContext.fetchCount(dslContext.selectFrom(this.quotedObjectTableSelectName)
-                                               .where(Tables.BITHON_ALERT_OBJECT.ALERT_ID.eq(alertId))
-                                               .and(Tables.BITHON_ALERT_OBJECT.DELETED.eq(0))) > 0;
-
+        return this.fetchCount(this.quotedObjectTableSelectName,
+                               Tables.BITHON_ALERT_OBJECT.ALERT_ID.eq(alertId),
+                               Tables.BITHON_ALERT_OBJECT.DELETED.eq(0)) > 0;
     }
 
     @Override
     public boolean existAlertByName(String name) {
-        return dslContext.fetchCount(dslContext.selectFrom(this.quotedObjectTableSelectName)
-                                               .where(Tables.BITHON_ALERT_OBJECT.ALERT_NAME.eq(name))
-                                               .and(Tables.BITHON_ALERT_OBJECT.DELETED.eq(0))) > 0;
-
+        return this.fetchCount(this.quotedObjectTableSelectName,
+                               Tables.BITHON_ALERT_OBJECT.ALERT_NAME.eq(name),
+                               Tables.BITHON_ALERT_OBJECT.DELETED.eq(0)) > 0;
     }
 
     @Override
@@ -255,16 +254,16 @@ public class AlertObjectJdbcStorage implements IAlertObjectStorage {
 
     @Override
     public int getAlertListSize(String appName, String alertName) {
-        SelectConditionStep<?> selectSql = dslContext.selectFrom(this.quotedObjectTableSelectName).where(Tables.BITHON_ALERT_OBJECT.DELETED.eq(0));
+        Condition condition = Tables.BITHON_ALERT_OBJECT.DELETED.eq(0);
 
         if (StringUtils.hasText(appName)) {
-            selectSql = selectSql.and(Tables.BITHON_ALERT_OBJECT.APP_NAME.eq(appName));
+            condition = condition.and(Tables.BITHON_ALERT_OBJECT.APP_NAME.eq(appName));
         }
         if (StringUtils.hasText(alertName)) {
-            selectSql = selectSql.and(Tables.BITHON_ALERT_OBJECT.ALERT_NAME.like("%" + alertName + "%"));
+            condition = condition.and(Tables.BITHON_ALERT_OBJECT.ALERT_NAME.like("%" + alertName + "%"));
         }
 
-        return dslContext.fetchCount(selectSql);
+        return this.fetchCount(this.quotedObjectTableSelectName, condition);
     }
 
     @Override
@@ -360,18 +359,25 @@ public class AlertObjectJdbcStorage implements IAlertObjectStorage {
 
     @Override
     public ListResult<AlertChangeLogObject> getChangeLogs(String alertId, Integer pageNumber, Integer pageSize) {
-        SelectConditionStep<?> selectSql = dslContext.selectFrom(Tables.BITHON_ALERT_CHANGE_LOG)
-                                                     .where(Tables.BITHON_ALERT_CHANGE_LOG.ALERT_ID.eq(alertId));
-
-        return new ListResult<>(dslContext.fetchCount(selectSql),
-                                selectSql.orderBy(Tables.BITHON_ALERT_CHANGE_LOG.CREATED_AT.desc())
-                                         .offset(pageNumber * pageSize)
-                                         .limit(pageSize)
-                                         .fetchInto(AlertChangeLogObject.class));
+        return new ListResult<>(dslContext.fetchCount(Tables.BITHON_ALERT_CHANGE_LOG, Tables.BITHON_ALERT_CHANGE_LOG.ALERT_ID.eq(alertId)),
+                                dslContext.selectFrom(Tables.BITHON_ALERT_CHANGE_LOG)
+                                          .where(Tables.BITHON_ALERT_CHANGE_LOG.ALERT_ID.eq(alertId))
+                                          .orderBy(Tables.BITHON_ALERT_CHANGE_LOG.CREATED_AT.desc())
+                                          .offset(pageNumber * pageSize)
+                                          .limit(pageSize)
+                                          .fetchInto(AlertChangeLogObject.class));
     }
 
     @Override
     public <T> T executeTransaction(Callable<T> callable) {
         return this.dslContext.transactionResult((configuration) -> callable.call());
+    }
+
+    private int fetchCount(String table, Condition... conditions) {
+        //noinspection DataFlowIssue
+        return dslContext.selectCount()
+                         .from(table)
+                         .where(conditions)
+                         .fetchOne(0, int.class);
     }
 }

--- a/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/AlertRecordJdbcStorage.java
+++ b/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/AlertRecordJdbcStorage.java
@@ -126,7 +126,8 @@ public class AlertRecordJdbcStorage implements IAlertRecordStorage {
         if (alertId != null) {
             sql = ((SelectWhereStep<?>) sql).where(Tables.BITHON_ALERT_RECORD.ALERT_ID.eq(alertId));
         }
-        return new ListResult<>(dslContext.fetchCount(sql),
+
+        return new ListResult<>(dslContext.selectCount().from(sql).fetchOne(0, int.class),
                                 ((SelectConditionStep<?>) sql).orderBy(Tables.BITHON_ALERT_RECORD.CREATED_AT.desc())
                                                               .limit(pageSize)
                                                               .offset(pageSize * pageNumber)

--- a/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/NotificationChannelJdbcStorage.java
+++ b/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/alerting/NotificationChannelJdbcStorage.java
@@ -72,8 +72,7 @@ public class NotificationChannelJdbcStorage implements IAlertNotificationChannel
 
     @Override
     public boolean exists(String name) {
-        return dslContext.fetchCount(dslContext.selectFrom(Tables.BITHON_ALERT_NOTIFICATION_CHANNEL)
-                                               .where(Tables.BITHON_ALERT_NOTIFICATION_CHANNEL.NAME.eq(name))) > 0;
+        return dslContext.fetchCount(Tables.BITHON_ALERT_NOTIFICATION_CHANNEL, Tables.BITHON_ALERT_NOTIFICATION_CHANNEL.NAME.eq(name)) > 0;
     }
 
     @Override

--- a/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/meta/SchemaJdbcStorage.java
+++ b/server/storage-jdbc/src/main/java/org/bithon/server/storage/jdbc/meta/SchemaJdbcStorage.java
@@ -86,12 +86,8 @@ public class SchemaJdbcStorage implements ISchemaStorage {
 
     @Override
     public boolean containsSchema(String name) {
-        // fetchExists API generates SQL as
-        // select 1 "one" where exists ( select ... from ... where name = '')
-        // such SQL is not supported by ClickHouse
-        return dslContext.fetchCount(dslContext.select(Tables.BITHON_META_SCHEMA.NAME)
-                                               .from(Tables.BITHON_META_SCHEMA)
-                                               .where(Tables.BITHON_META_SCHEMA.NAME.eq(name))) > 0;
+        return dslContext.fetchCount(Tables.BITHON_META_SCHEMA,
+                                     Tables.BITHON_META_SCHEMA.NAME.eq(name)) > 0;
     }
 
     @Override

--- a/server/web-app/src/main/resources/templates/alerting/record/records.html
+++ b/server/web-app/src/main/resources/templates/alerting/record/records.html
@@ -12,7 +12,6 @@
 
     <script th:src="@{/lib/echarts5.0.1/echarts.min.js}"></script>
     <script th:src="@{/js/chart-component.js}"></script>
-    <script th:src="@{/js/component/tree-table/component.js}"></script>
     <script th:src="@{/js/utils/location.js}"></script>
 
     <style>

